macros/contexts/contextReaction.pl

```diff
@@ -744,10 +749,12 @@ sub string {
 #  Always put signs on the right
 #
 sub TeX {
-	my $self = shift;
-	my $uop  = $self->{def};
-	my $op   = (defined($uop->{TeX}) ? $uop->{TeX} : $uop->{string});
-	return $self->{op}->TeX($uop->{precedence}) . $op;
+	my $self      = shift;
+	my $uop       = $self->{def};
+	my $op        = (defined($uop->{TeX}) ? $uop->{TeX} : $uop->{string});
+	my $magnitude = $self->{op}->TeX($uop->{precedence});
+	return $op if $self->context->flags->get('showUnity') == 0 && $magnitude == 1;
```

Comment:
Since `$magnitude` is a string, I would recommend using `$magnitute eq '1'` instead of the numeric comparison.  Although that works, you can avoid making the numeric conversion from the string that way.

Also, note that the `string()` method above (lines 742 to 746 in your file) already does the removal of the `1`, so you should probably work your `showUnity` check into that as well.

Finally, I would recommend using `!$self->context->flags->get('showUnity')` rather than a test specifically against 0, since that will cover any false-like value (and your test will produce true if `showUnity` were set to some non-numeric string, since that will convert to 0 for the numeric test).